### DISCUSSION TITLE
Why direct properties don't work?

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#6279] [Q&A]
I tried to add custom property for my user control, but something went wrong so it didn't set SetText property.
[InputField.axaml - user controls](https://github.com/AvaloniaUI/Avalonia/files/6845456/InputField.axaml.txt)
[InputField.axaml - code for user controls](https://github.com/AvaloniaUI/Avalonia/files/6845474/InputField.axaml.txt)

Using in window
xmlns:view="clr-namespace:VkUtilityAvalonia"
...
<view:InputField SetText="Text"></view:InputField>
...

**Comment by maxkatz6** [comment]
Wrong property name. It should be "Text" everywhere, if you defined it as "Text" in direct property. Not "SetText"

**Reply by ghost** [reply]
Oh, sorry, its right. At this morning I'm tried to change it because i thought that Text override Text in textBlock may be, but it doesn't help me at all.

**Reply by ghost** [reply]
Anyways, it still don't work in here and on my computer.

**Reply by maxkatz6** [reply]
Need more details, updated code, better small project to reproduce.

**Reply by ghost** [reply]
Made repository for this purposes.

https://github.com/PixelGuy19/VkUtilityAvalonia

**Reply by maxkatz6** [reply]
I see why direct property does not work.
You copy value of "SetText" to InputFieldViewModel before ```SetText="Text"``` was executed.
You need to remember that xaml code:
```
<view:InputField SetText="Text"></view:InputField>
```
Works in similar way as:
```
var inputField = new InputField();
inputField.SetText = "Text";
```

And in your code, you create InputFieldViewModel with default SetText value in InputField ctor, before SetText was set to another value.
